shared/github_oauth.go

```diff
@@ -174,20 +193,33 @@ func NewGitHubAccessControl(ctx context.Context, ds Datastore, botClient *github
 	}, nil
 }
 
-// GetSecureCookie returns the securecookie instance for wpt.fyi. This instance can
-// be used to encode and decode cookies set by wpt.fyi.
-func GetSecureCookie(ctx context.Context, store Datastore) (*securecookie.SecureCookie, error) {
-	log := GetLogger(ctx)
-	hashKey, err := GetSecret(store, "secure-cookie-hashkey")
+// NewGitHubAccessControlFromRequest returns a GitHubAccessControl for checking
+// the permission of a logged-in GitHub user from a request. (nil, nil) will be
+// returned if the user is not logged in.
+func NewGitHubAccessControlFromRequest(aeAPI AppEngineAPI, ds Datastore, r *http.Request) (GitHubAccessControl, error) {
+	ctx := aeAPI.Context()
+	botClient, err := aeAPI.GetGitHubClient()
 	if err != nil {
-		log.Errorf("Failed to get secure-cookie-hashkey secret: %s", err.Error())
 		return nil, err
 	}
+	user, token := GetUserFromCookie(ctx, ds, r)
+	if user == nil {
+		return nil, nil
```

Comment:
This should return an `error`

Reply:
Hmm, I'm trying to reserve errors for unexpected stuff (my response to your comment above is in the same vein). In this case, I think missing (valid) cookies is expected (user not logged in) so it shouldn't be an error. WDYT?

Reply:
Yea that makes sense. Also see [my comment below](https://github.com/web-platform-tests/wpt.fyi/pull/2175/files#r495681167)